cli.py

```diff
@@ -0,0 +1,166 @@
+        "hash",
+        help="Compute a 256bit Pedersen hash. Preimage size is set to 512bit as default",
+    )
+    pedersen_parser.add_argument(
+        "preimage", nargs=1, help="Provide preimage as hexstring"
+    )
+    pedersen_parser.add_argument(
+        "-s", "--size", type=int, help="Define message size in bits", default=64
+    )
+    pedersen_parser.add_argument(
+        "-p", "--personalisation", help="Provide personalisation string", default="test"
+    )
+
+    # batch pedersen hash subcommand
+    pedersen_hasher_parser = subparsers.add_parser(
+        "batch_hasher",
+        help="Efficiently compute multiple Pedersen hashes. Support for stdin and alive promt",
+    )
+
+    pedersen_hasher_parser.add_argument(
+        "-s", "--size", type=int, help="Define message size in bits", default=64
+    )
+    pedersen_hasher_parser.add_argument(
+        "-p", "--personalisation", help="Provide personalisation string", default="test"
+    )
+
+    # keygen subcommand
+    keygen_parser = subparsers.add_parser(
+        "keygen",
+        help="Returns space separated hex-string for a random private/public keypair on BabyJubJub curve",
+    )
+    keygen_parser.add_argument(
+        "-p",
+        "--from_private",
+        help="Provide existing private key as hexstring (64 chars)",
+    )
+
+    # eddsa signature generation subcommand
+    sig_gen_parser = subparsers.add_parser(
+        "sig-gen",
+        help="Returns eddsa signature as space separated hex-string. Private key and message needs to be provided",
+    )
+    sig_gen_parser.add_argument(
+        "private_key", nargs=1, help="Provide public key as hexstring (64chars)"
+    )
+
+    sig_gen_parser.add_argument(
+        "message", nargs=1, help="Provide message as string, will be utf-8 encoded"
+    )
+
+    # eddsa signature verify subcommand
+    sig_verify_parser = subparsers.add_parser(
+        "sig-verify", help="Verifies a eddsa signaure. Returns boolean flag for success"
+    )
+    sig_verify_parser.add_argument(
+        "public_key", nargs=1, help="Provide public key as hexstring (64chars)"
+    )
+    sig_verify_parser.add_argument(
+        "message", nargs=1, help="Provide message as string, will be utf-8 encoded"
+    )
+    sig_verify_parser.add_argument(
+        "signature",
+        nargs=2,
+        help="Provide signaure as space separated hexsting (2x 64 chars)",
+    )
+
+    args = parser.parse_args()
+    subparser_name = args.subparser_name
+
+    if subparser_name == "hash":
+        preimage = bytes.fromhex(args.preimage[0])
+        if len(preimage) != args.size:
+            raise ValueError(
+                "Bad lenght for preimage: {} vs {}".format(len(preimage), args.size)
+            )
+
+        personalisation = args.personalisation.encode("ascii")
+        point = PedersenHasher(personalisation).hash_bytes(preimage)
+        digest = point.compress()
+
+        assert len(digest.hex()) == 32 * 2  # compare to hex string
```

Comment:
Might be just me, but I'm not sure I understand the comment here. Could be worth clarifying/expanding.